### DISCUSSION TITLE
automatic commands regestration | автоматическое регестрирование команд

### DISCUSSION THREAD

**Discussion by yoteamomiamor** [aiogram/aiogram#1505] [Ideas]
I suggest a function for automatic addition of commands to the bot menu. The idea is that if the command handlers contain a docstring, they should automatically register in the menu, like when calling bot.set_my_commands(), but without the need to manually specify them.

The only point - if there are multiple handlers for one command (e.g. in different states or chats), we can use the first registered one.

Overall, this could simplify the bot menu setup and make the process more automated. We need to work out the details, but the idea seems interesting.

-----------------------------------------------------------------------------

Предлагаю добавить функцию автоматического добавления команд в меню бота. Идея в том, что если хендлеры команд содержат docstring, то они должны автоматически регистрироваться в меню, как при вызове bot.set_my_commands(), но без необходимости это вручную прописывать.

Единственный момент - если для одной команды есть несколько хендлеров (например, в разных состояниях или чатах), то можно использовать первый зарегистрированный.

В целом, это может упростить настройку меню бота и сделать процесс более автоматизированным. Нужно проработать детали, но идея интересная.

**Comment by JrooTJunior** [comment]
You can implement it now without any additional pre-requisites. We already has all required mechanisms to do it.

Read more about collecting meta-information about registered commands in the documentation: https://docs.aiogram.dev/en/dev-3.x/dispatcher/flags.html#use-in-utilities

But this functionality depends on project specification, in some cases we don't need to have all commands in the commands menu or some of the commands should be available only for administrators (should be added with specific scope), so, we can't add deeper integration than it is now or please provide code examples how it can looks like (not implementation, just usage).
